File: sorting_algorithms/insertion_sort/insertion_sort.cpp
#include <vector>
#include <algorithm>

template<typename T>
void insertion_sort(std::vector<T>& vector) {
    if (vector.size() <= 1) {
        return;
    }
    
    for (size_t iteration = 1; iteration < vector.size(); iteration++) {
        for (size_t index = iteration; index >= 1; index--) {
            if (vector[index - 1] < vector[index]) {
                std::swap(vector[index - 1], vector[index]);
            } else {
                break;
            }
        }
    }
}

File: sorting_algorithms/insertion_sort/insertion_sort.py

def insertion_sort(vector: list):
    for iteration in range(len(vector)):
        for index in range(iteration, 0, -1):
            if vector[index - 1] < vector[index]:
                vector[index], vector[index - 1] = vector[index - 1], vector[index]
            else:
                break